### PR TITLE
Fix bevy_ecs README.md 

### PR DESCRIPTION
**Pull request by redwarp**
# Objective

- The README.md for bevy_ecs contains an example for building a system that would not work as is. This change addresses that.

## Solution

- For using iter_mut(), the query parameter in the system function needs to be mut as well.
- For adding a fun as a system, `system()` needs to be called.

### CODE REVIEW

File: crates/bevy_ecs/README.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ fn main() {
     // Add a Stage to our schedule. Each Stage in a schedule runs all of its systems
     // before moving on to the next Stage
     schedule.add_stage("update", SystemStage::parallel()
-        .with_system(movement)
+        .with_system(movement.system())
```
Comment (cart) [inline crates/bevy_ecs/README.md:134]: `.system()` is not required on `main`.

Comment (alice-i-cecile) [inline crates/bevy_ecs/README.md:134]: No longer needed on `main`, see https://github.com/bevyengine/bevy/pull/2398

**Comment by DJMcNab** [comment]
Duplicate of #2575, #3083, #3100.

See also https://github.com/bevyengine/bevy/pull/2398, unless you're claiming it regressed.

**Comment by cart** [comment]
Yup adding `.system()` isn't necessary, but adding `mut` to the query variable is still needed, so we should merge that.

**Comment by redwarp** [comment]
Ah, I see, and as I'm using bevy_ecs 0.5, the patch from june is not live yet, gotcha. Will amend to just show the mut then.

**Comment by DJMcNab** [comment]
Indeed, although that fix is already in #2575

**Comment by redwarp** [comment]
@DJMcNab indeed, closing it then.

**Pull request closed without merging**